esp/services/ws_packageprocess/ws_packageprocessService.hpp

```diff
@@ -104,16 +104,24 @@ class PackageMapAndSet : public CInterface, implements ISDSSubscription
         return LINK(tree);
     }
 
+    IPropertyTree *getPackageMapsOrSets(IPropertyTree *root, const char* path)
+    {
+        Owned<IPropertyTree> ret = root->getPropTree(path);
+        if (!ret)
+            ret.setown(createPTree(path, ipt_lowmem));
+        return ret.getClear();
+    }
+
     IPropertyTree *getPackageMaps()
```

Comment:
I would probably rename these functions too since they now ensure the tree exists: "ensurePackageMaps()".